### DISCUSSION TITLE
node-gyp dependency vs prebuild on Windows

### DISCUSSION THREAD

**Discussion by dfgordon** [tree-sitter/tree-sitter#3297] [Q&A]
I found my language only works with node bindings if I include `node-gyp` as a dependency, but only on Windows.  On Linux this dependency was not needed.

On the other hand `tree-sitter-javascript` does not seem to require `node-gyp`, on Windows or otherwise.  The only difference I could find was that `tree-sitter-javascript` has the prebuilds directory, while my language does not.

Is it correct to conclude, as seems to make sense, that if prebuilds are available you don't need `node-gyp`, otherwise you do?  What I can't figure is why Linux didn't need it; maybe to do with specific config.

**Comment by amaanq** [comment]
I can't remember when I updated the javascript grammar (if it was before or after moving to NAPI and removing the need for node-gyp), but you do not need `node-gyp`, you can just use `prebuildify` or `node-gyp-build` to take care of that

**Reply by dfgordon** [reply]
So I have `prebuildify` and `node-gyp-build` in the package.json in what seems to be the same way as others.  Maybe the problem is the publishing steps.  I was supposing `tree-sitter-generate` would generate whatever is needed for publishing to npmjs.  Maybe prebuildify has to be run explicitly, is that right?  I'm not using the github workflows for publishing at present.

**Reply by amaanq** [reply]
if you have a scanner, you need to add this to binding.gyp otherwise it won't build properly

otherwise, yeah, `tree-sitter generate` adds the binding files you need, but you have to build them with prebuildify
